WalletWasabi/Blockchain/Analysis/FeesEstimation/PrivacyFeeProvider.cs

```diff
@@ -38,24 +38,29 @@ public FeeProviders(IFeeProvider baseProvider, params IFeeProvider[]? feeProvide
 
 		private void Provider_AllFeeEstimateChanged(object? sender, AllFeeEstimate e)
 		{
-			SetAllFeeEstimate();
+			SetEstimate();
 		}
 
-		private void SetAllFeeEstimate()
+		private void SetEstimate()
 		{
 			lock (Lock)
 			{
-				AllFeeEstimate? feeEstimateToSet = null;
+				AllFeeEstimate? bestProviderEstimate = null;
 				foreach (IFeeProvider provider in Providers.SkipLast(1))
 				{
 					if (provider.AllFeeEstimate is { IsAccurate: bool isAccurate } af && isAccurate)
 					{
-						feeEstimateToSet = af;
+						bestProviderEstimate = af;
 						break;
 					}
 				}
+				bestProviderEstimate ??= Providers.Last().AllFeeEstimate;
 
-				AllFeeEstimate = feeEstimateToSet ?? Providers.Last().AllFeeEstimate;
+				var privacyEstimate = bestProviderEstimate?.Unfingerprint();
+				if (privacyEstimate is { })
+				{
+					AllFeeEstimate = privacyEstimate;
```

Comment:
`AllFeeEstimate` is not assigned always now, so `AllFeeEstimateChanged?.Invoke(this, AllFeeEstimate);` can use old value?